topic/timeseries/timeseries-queries-and-visualization.ipynb

```diff
@@ -439548,28 +439661,74 @@
+    "In the same spirit as Lotus Notes/Domino, CouchDB, and MongoDB, CrateDB offers a powerful\n",
+    "data type suitable for storing and retrieving \"documents\", mostly conceptualized as\n",
+    "»a bunch of JSON data« attached to a data point.\n",
+    "\n",
+    "In this manner, a single data point / record isn't restricted to name/label/tag/value fields\n",
+    "only, like what popular metrics aggregation and monitoring systems with corresponding\n",
+    "timeseries databases can offer.\n",
+    "\n",
+    "In CrateDB jargon, it is the [`OBJECT` datatype], which also implements three different\n",
+    "policies (`DYNAMIC`, `STRICT`, or `IGNORED`), and defaults to `DYNAMIC`, which effectively\n",
+    "means \"store just anything and expand the schema as we go\".\n",
+    "\n",
+    "[`OBJECT` datatype]: https://cratedb.com/docs/crate/reference/en/latest/general/ddl/data-types.html#objects"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Prepare the Database Schema (DDL)\n",
+    "\n",
+    "Applied to the earth observations example at hand, imagine the data you insert can vary per\n",
+    "sensor type / weather station, so it needs to follow a dynamic schema. Learning from the\n",
+    "elaborations above, it is clear that `OBJECT(DYNAMIC)` is the right choice for a new\n",
+    "database column `weather_data`."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
   {
    "cell_type": "code",
-   "execution_count": 11,
+   "execution_count": 0,
    "id": "73c2fcab",
    "metadata": {},
    "outputs": [],
    "source": [
-    "# Add an additional column that holds the dynamic data.\n",
-    "result = connection.execute(sa.text('ALTER TABLE weather_data ADD COLUMN attributes OBJECT(DYNAMIC)'))\n",
+    "# Add a new column that stores the dynamic data.\n",
+    "result = connection.execute(sa.text('ALTER TABLE weather_data ADD COLUMN attributes OBJECT(DYNAMIC)'))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Insert Records with Documents\n",
     "\n",
-    "# Use arbitrary JSON directly in the INSERT statement\n",
+    "The code snippets below demonstrate how to insert a nested JSON document into a\n",
+    "database record."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "outputs": [],
+   "source": [
+    "# Use arbitrary JSON directly within the INSERT statement.\n",
     "result = connection.execute(sa.text(\n",
     "    \"\"\"\n",
     "    INSERT INTO weather_data (timestamp, location, temperature, humidity, wind_speed, attributes) VALUES (\n",
-    "    CURRENT_TIMESTAMP, 'Vienna', 9.2, 80, 7.2, '{\"measurement_accuracy\": \"+/- 0.5°C\",\n",
+    "    CURRENT_TIMESTAMP, 'Vienna', 9.2, 80, 7.2,\n",
+    "      '{\"measurement_accuracy\": \"+/- 0.5°C\",\n",
```

Comment:
I've wrapped a line here within a code cell, without verifying it. Please make sure it doesn't break until we added software tests.